### ISSUE TITLE
Autocorrect option fixes hash in an ugly way

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#11900]
## Actual behavior

I have the code:

```ruby
specify do
  expect(result).to eq({
    campaign1.id => rand(1..10),
    campaign2.id => rand(1..10),
  })
end
```

I run rubocop with the autocorrect option:

```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -A spec/presenter_spec.rb.rb
Inspecting 1 file
C

Offenses:

spec/presenter_spec.rb.rb:181:13: C: [Corrected] Layout/FirstHashElementIndentation: Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.
            campaign1.id => rand(1..10),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
spec/presenter_spec.rb.rb:182:13: C: [Corrected] Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
            campaign2.id => rand(1..10),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
spec/presenter_spec.rb.rb:183:11: C: [Corrected] Layout/FirstHashElementIndentation: Indent the right brace the same as the first position after the preceding left parenthesis.
          })
          ^
```

I got the code:

```ruby
specify do
  expect(result).to eq({
                         campaign1.id => rand(1..10),
                         campaign2.id => rand(1..10),
                       })
end
```

## Expected behavior

But I expected to receive the code:

```ruby
specify do
  expect(result).to eq(
    campaign1.id => rand(1..10),
    campaign2.id => rand(1..10),
  )
end
```

## RuboCop version

```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.50.2 (using Parser 3.2.2.1, rubocop-ast 1.28.1, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.18.0
  - rubocop-performance 1.17.1
  - rubocop-rails 2.19.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.22.0
  - rubocop-thread_safety 0.5.1
```

**Comment by naveg** [comment]
Try looking into the `EnforcedStyle: consistent` option for cops that support it.

https://docs.rubocop.org/rubocop/cops_layout.html#enforcedstyle-consistent-3

**Comment by ydakuka** [comment]
> Try looking into the EnforcedStyle: consistent option for cops that support it.

I've tried it and rubocop hasn't found any offences. And this cop don't remove the redundant parentheses at all.

**Comment by koic** [comment]
`Style/BracesAroundHashParameters` cop has been removed in https://github.com/rubocop/rubocop/pull/7643 to accommodate the separation of real keyword arguments in Ruby 3.0. RuboCop's static analysis cannot determine whether or not braces can be reliably omitted.

**Comment by ydakuka** [comment]
@koic thank you.
